File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int i = 0;
        while(i< nums.length){
            int correct = nums[i];
            if (nums[i]<nums.length && nums[i] != nums[correct]){
                swap(nums, i, correct);
            } else{
                i++;
            }

        }
        for (int j = 0; j < nums.length; j++) {
                if(j!=nums[j]){
                    return j;
                }

            }
        return nums.length;
    }
    void swap(int[] arr, int first, int second){
        int temp = arr[first];
        arr[first] = arr[second];
        arr[second] = temp;
    }
}

File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        # return sum(range(len(nums) + 1)) - sum(nums) # One line version
        actualSum = 0
        for i in range(len(nums)):
            actualSum += nums[i]
            
        expectedSum = 0
        for i in range(len(nums)+1):
            expectedSum += i
            
        return expectedSum - actualSum
            